File: frontend/src/types/case.ts
/** Domains supported by the legal engine. */
export type Domain = "consumer" | "labor" | "tenant" | "cyber" | "criminal" | "general";

export type Language = "en" | "hi";

export type Urgency = "low" | "medium" | "high" | "urgent";

/** Lifecycle status persisted on the case. String-open to tolerate backend additions. */
export type CaseStatus =
  | "new"
  | "classified"
  | "retrieved"
  | "clarified"
  | "explained"
  | "documented"
  | (string & {});

/** Structured facts extracted from the user's description (mirrors CaseFactsData). */
export interface CaseFacts {
  parties?: string | null;
  incident?: string | null;
  date?: string | null;
  location?: string | null;
  state?: string | null;
  subdomain?: string | null;
  amount?: string | number | null;
  agreement_exists?: boolean | null;
  notice_given?: boolean | null;
  desired_outcome?: string | null;
  urgency?: Urgency | null;
  additional_facts?: string | null;
}

/** POST /api/cases */
export interface CreateCaseRequest {
  text: string;
  language?: Language;
  session_id?: string;
}

export interface CreateCaseResponse {
  case_id: string;
  language: string;
  status: string;
  created_at?: string;
}

/** GET /api/cases/{id} */
export interface CaseState {
  case_id: string;
  session_id?: string | null;
  language: string;
  original_text: string;
  domain?: Domain | null;
  subdomain?: string | null;
  state?: string | null;
  urgency: Urgency;
  status: CaseStatus;
  created_at: string;
  facts?: CaseFacts | null;
}

File: backend/app/schemas/case.py
"""Pydantic schemas for Case intake and session status."""

from typing import Optional, Dict, Any, List
from pydantic import BaseModel, Field


class CreateCaseRequest(BaseModel):
    text: str = Field(..., min_length=1, description="User's legal problem description (Hindi or English)")
    language: str = Field(default="en", description="Language code: 'en' or 'hi'")
    session_id: Optional[str] = Field(None, description="Optional session tracking identifier")


class CreateCaseResponseData(BaseModel):
    case_id: str
    language: str
    status: str


class CaseFactsData(BaseModel):
    parties: Optional[str] = None
    incident: Optional[str] = None
    date: Optional[str] = None
    location: Optional[str] = None
    state: Optional[str] = None
    subdomain: Optional[str] = None
    amount: Optional[str] = None
    agreement_exists: Optional[bool] = None
    notice_given: Optional[bool] = None
    desired_outcome: Optional[str] = None
    urgency: Optional[str] = "low"
    additional_facts: Optional[Dict[str, Any]] = None


class CaseStateData(BaseModel):
    case_id: str
    session_id: Optional[str] = None
    language: str
    original_text: str
    domain: Optional[str] = None
    subdomain: Optional[str] = None
    state: Optional[str] = None
    urgency: Optional[str] = "low"
    status: str
    created_at: str
    facts: Optional[CaseFactsData] = None
